src/locksManager.js

```diff
@@ -54,4 +54,20 @@ var LocksManager = GObject.registerClass({
         const variant = new GLib.Variant('a{sb}', {locked: false});
         this._proxy.SetSync(lock, variant);
     }
+
+    getTrapSequence(lock) {
+        return this._proxy.getDictValueSync(lock, 'trap_sequence');
+    }
+
+    setLockTried(lock, tried) {
+        try {
+            const [variant] = this._proxy.GetSync(lock);
+            const dict = new GLib.VariantDict(variant);
```

Comment:
I am pretty sure this can be done with `new GLib.Variant('a{sv}', {tried: new GLib.Variant('b', tried)});`

Reply:
I tried your suggestion using a new variant and it didn't work. What this does is update an existing lock. I did it in the same way as setUsed above in the file. So for example:
```
  "lock.OperatingSystemApp.2" : {
    "locked" : false
  }
```
Becomes:
```
  "lock.OperatingSystemApp.2" : {
    "locked" : false,
    "tried" : true
  }
```

Reply:
Oh, I see, it's the equivalent of `dict.update()` in Python. Never mind my comment then